### PR TITLE
Allow forcing a Block Index Update via GraphQL Query

### PR DESCRIPTION
**Pull request by thomasdax98**
depends on https://github.com/vivid-planet/comet/pull/1222

- add a `forceRefresh` arg to `dependencies()` and `dependents()` field resolver

### CODE REVIEW

File: packages/api/cms-api/src/dependencies/dto/dependencies.args.ts [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
 @ArgsType()
 export class DependentsArgs extends OffsetBasedPaginationArgs {
     @Field(() => DependentFilter, { nullable: true })
     @ValidateNested()
     @Type(() => DependentFilter)
-    @IsOptional()
+    @IsUndefinable()
     filter?: DependentFilter;
+
+    @Field(() => Boolean, { nullable: true, defaultValue: false })
```
Comment (nsams) [inline packages/api/cms-api/src/dependencies/dto/dependencies.args.ts:30]: doesn't have to be nullable if you specify a defaultValue

File: packages/api/cms-api/src/dependencies/dto/dependencies.args.ts [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
+    @IsUndefinable()
+    force: boolean;
```
Comment (nsams) [inline packages/api/cms-api/src/dependencies/dto/dependencies.args.ts:32]: force in the graphql api is a bit strange parameter name, alternatives:

- forceRefresh
- refresh
- update
- forceUpdate

**Review by johnnyomair — COMMENTED** [review]
Is it a good idea to do this as a side effect of the dependents/dependencies queries? Shouldn't this be a separate mutation, for instance, refreshDependencies?

**Comment by nsams** [comment]
> Is it a good idea to do this as a side effect of the dependents/dependencies queries? Shouldn't this be a separate mutation, for instance, refreshDependencies?

I think it is Ok. With a mutation you win nothing only complicate things.

File: packages/api/cms-api/src/dependencies/dto/dependencies.args.ts [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
 export class DependentsArgs extends OffsetBasedPaginationArgs {
     @Field(() => DependentFilter, { nullable: true })
     @ValidateNested()
     @Type(() => DependentFilter)
-    @IsOptional()
+    @IsUndefinable()
     filter?: DependentFilter;
+
+    @Field(() => Boolean, { defaultValue: false })
+    @IsUndefinable()
```
Comment (nsams) [inline packages/api/cms-api/src/dependencies/dto/dependencies.args.ts:31]:
```suggestion
```

File: packages/api/cms-api/src/dependencies/dto/dependencies.args.ts [hunk 4]
```diff
@@ -1,24 +1,33 @@
...
 export class DependenciesArgs extends OffsetBasedPaginationArgs {
     @Field(() => DependencyFilter, { nullable: true })
     @ValidateNested()
     @Type(() => DependencyFilter)
-    @IsOptional()
+    @IsUndefinable()
     filter?: DependencyFilter;
+
+    @Field(() => Boolean, { defaultValue: false })
+    @IsUndefinable()
```
Comment (johnnyomair) [inline packages/api/cms-api/src/dependencies/dto/dependencies.args.ts:18]:
```suggestion
```

**Review by johnnyomair — CHANGES_REQUESTED** [review]

**Review by nsams — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**